### PR TITLE
fixed query

### PR DESCRIPTION
**Pull request by xrotwang**

### CODE REVIEW

File: etc/views.sql [hunk 1]
```diff
@@ -49,10 +49,11 @@ GROUP BY w.cldf_languageReference;
...
 SELECT                                                             
 	w.cldf_languageReference,
+	w.cldf_name,
 	-- word form frequency
-	count(w.cldf_name) as WordFreq
+	count(w.cldf_id) as WordFreq
 FROM
 	'words.csv' as w
-GROUP BY w.cldf_languageReference, cldf_name;
+GROUP BY w.cldf_languageReference, w.cldf_name;
```
Comment (xrotwang) [inline etc/views.sql:57]: @Tarotis I suppose that's what we want here, right? Number of words per (form, language)?

File: init_query.sql [hunk 2]
```diff
@@ -14,7 +14,7 @@ SELECT
         WHEN sound.cldf_cltsReference like '%stop%' then 'stop' else 'dunno'
         END sound_class,
 	-- standardize speech rate
-	-- (utt.log_speech_rate - avg(utt.log_speech_rate) over()) / stdev(utt.log_speech_rate) over() as z_speech_rate,
+	(utt.log_speech_rate - z.avg_speech_rate) / z.speech_rate AS z_speech_rate,
```
Comment (xrotwang) [inline init_query.sql:17]: @Tarotis we can pull average and stdev (computed over all utterances) from the joined query `z`, see below.

File: init_query.sql [hunk 3]
```diff
@@ -43,8 +43,13 @@ LEFT JOIN
     utterances AS utt  -- utterance-level stats such as speech rate.
 ON
     phone.u_ID = utt.u_id
+LEFT JOIN
+	(
+	    SELECT stdev(u.log_speech_rate) AS speech_rate, AVG(u.log_speech_rate) AS avg_speech_rate FROM utterances as u
+    ) AS z
```
Comment (xrotwang) [inline init_query.sql:49]: @Tarotis We compute average and stdev over all utterances here.

File: init_query.sql [hunk 4]
```diff
@@ -43,8 +43,13 @@ LEFT JOIN
...
 WHERE
-    ws.cldf_languageReference = word.cldf_languageReference AND
+    (ws.cldf_languageReference = word.cldf_languageReference AND ws.cldf_name = word.cldf_name) AND
+    ls.cldf_languageReference = word.cldf_languageReference AND
```
Comment (xrotwang) [inline init_query.sql:52]: The `WHERE` clause condition for the `langstats` view was missing. I suppose that was the issue blowing up the query on your laptop.

**Comment by xrotwang** [comment]
@Tarotis `init_query.sql` as it is now runs in a little less than 2 mins on my machine (resulting in 1115153 rows).

File: etc/views.sql (hunk 1 above)
Comment (FredericBlum) [reply etc/views.sql:57]: ups, yes, I forgot the w. because I was trying out the code within the table itself

File: init_query.sql (hunk 3 above)
Comment (FredericBlum) [reply init_query.sql:49]: perfect, I can implement this for the other cases (word form frequency) as well I guess

File: init_query.sql (hunk 4 above)
Comment (FredericBlum) [reply init_query.sql:52]: Ah, I didn't realize I need the where clause. I need to read up a little bit on that

**Comment by FredericBlum** [comment]
Great, thank you! :)

File: init_query.sql (hunk 4 above)
Comment (xrotwang) [reply init_query.sql:52]: There's a bit of a philosophical discussion about whether to join explicitly or implicitly (see https://dataschool.com/how-to-teach-people-sql/difference-between-where-and-on-in-sql/). I got a bit accustomed to use "implicit joins", i.e. `SELECT FROM a, b WHERE a.x = b.y`, but it seems - and you forgetting the WHERE clause might be evidence for this - that explicit joins are more transparent. Normally, I'm a fan of the "principle of locality", so I might switch to explicit joins, where the `ON` clause follows right away in the future.

**Pull request merged**